code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,249 @@
+		if(user)
+			to_chat(user, "<span class='warning'>[src] only accepts empty syringes.</span>")
+		return FALSE
+	var/in_clip = length(syringes) + (chambered?.BB ? 1 : 0)
+	if(in_clip < max_syringes)
+		if(user)
+			if(!user.unEquip(new_syringe))
+				return
+			to_chat(user, "<span class='notice'>You load \the [new_syringe] into [src].</span>")
+		syringes.Add(new_syringe)
+		new_syringe.loc = src
+		process_chamber() // Chamber the syringe if none is already
+		return TRUE
+	else
+		if(user)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+		return FALSE
+
+/obj/item/gun/syringe/rapidsyringe/attackby(obj/item/A, mob/user, params, show_msg)
+
+	if(istype(A, /obj/item/storage))
+		// Boxes can be dumped in.
+		var/obj/item/storage/container = A
+		if(!length(container.contents))
+			to_chat(user, "<span class='warning'>[A] is empty!</span>")
+			return TRUE
+		if(length(syringes) + (chambered?.BB ? 1 : 0) == max_syringes)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+			return TRUE
+
+		var/total_inserted = 0
+		for(var/obj/item/reagent_containers/syringe/S in container)
+			if((length(syringes) + (chambered?.BB ? 1 : 0)) == max_syringes)
+				// full
+				break
+
+			// only allow empty syringes
+			if(!S.reagents.total_volume)
+				syringes += S
+				S.loc = src
+				total_inserted++
+
+		if(total_inserted)
+			to_chat(user, "<span class='notice'>You load [total_inserted] empty syringes into [src].")
+			// Chamber a syringe.
+			process_chamber()
+
+	else if(istype(A, /obj/item/reagent_containers/syringe))
+		insert_single_syringe(A, user)
+
+	else if(istype(A, /obj/item/reagent_containers))
+		var/obj/item/reagent_containers/incoming = A
+		if(!incoming.reagents.total_volume)
+			to_chat(user, "<span class='warning'>[incoming] is empty!</span>")
+			return
+
+		if(internal_beaker.reagents.holder_full())
+			to_chat(user, "<span class='warning'>[src]'s internal reservoir is full!</span>")
+			return
+
+		var/trans = incoming.reagents.trans_to(internal_beaker, incoming.amount_per_transfer_from_this)
+		to_chat(user, "<span class='notice'>You transfer [trans] unit\s of the solution to [src]'s internal reservoir.</span>")
+		update_loaded_syringe()
+
+	else
+		return ..()
+
+// Allow for emptying your deathmix, or for sec to find out what you were dumping into people
+/obj/item/gun/syringe/rapidsyringe/afterattack(atom/target, mob/living/user, flag, params)
+	if(istype(target, /obj/item/reagent_containers) && !istype(target, /obj/item/reagent_containers/syringe))
+		var/obj/item/reagent_containers/destination = target
+		if(!internal_beaker.reagents.total_volume)
+			to_chat(user, "<span class='warning'>[src]'s internal reservoir is empty!</span>")
+			return
+
+		if(destination.reagents.holder_full())
+			to_chat(user, "<span class='warning'>[destination] is full.</span>")
+			return
+
+		var/trans = internal_beaker.reagents.trans_to(destination, internal_beaker.amount_per_transfer_from_this)
```

Comment:
you can never change the internal beakers transfer rate, so theres not much point in having this. I feel like it should just be `trans_to(destination, 15)` or, make use of `get_units_per_shot()`

Reply:
True, yeah. I think I copy-pasted from the beaker code itself.